### ISSUE TITLE
MapDB corruption - java.lang.AssertionError: unknown trans log instruction '0'

### ISSUE THREAD

**Issue by mmarin-m** [jankotek/MapDB#369]
Hi.

This is a very similar or the same issue describe in [#359](https://github.com/jankotek/MapDB/issues/359).
I am using the 1.0.6 mapdb version, 1.0.7 Java version and windows and I got this error the second time that I perform the commit to disk.

```
java.lang.AssertionError: unknown trans log instruction '0' at log offset: 182321185
	at org.mapdb.StoreWAL.replayLogFile(StoreWAL.java:858)
	at org.mapdb.StoreWAL.commit(StoreWAL.java:637)
	at org.mapdb.EngineWrapper.commit(EngineWrapper.java:94)
	at org.mapdb.EngineWrapper.commit(EngineWrapper.java:94)
	at org.mapdb.DB.commit(DB.java:1595)
```
 
The first time seems to work fine and create a file of around 600 mb. 

We are trying to store around 50.000 users (strings) with 100 groups (strings).each one.
We are using two mapDBs one in memory, other in disk.

```
 public UserGroupMapDB() throws Exception {
    try {
      // temp db
      cache = DBMaker.newTempTreeSet();
      keys = DBMaker.newTempHashSet();
    }
    catch (Throwable t) {
      throw new Exception("UserGroupMapDB.database-creation-error", t, "Could not create database");
    }
  }
  public UserGroupMapDB(String name) throws Exception {
    try {
      // permanent db
      FileUtilities.mkdirsForFile(name);
      db = DBMaker.newFileDB(new File(name)).closeOnJvmShutdown().make();
      cache = db.getTreeSet("userGroups");
      keys = db.getHashSet("users");
    }
    catch (Throwable t) {
      throw new Exception("UserGroupMapDB.database-creation-error", t, "Could not create database: %s", name);
    }
  }
```

The permanetDB can be read any time, so we use the tempDB to perform the process to fill the data and later we swapped the entries of the tempDB with the permanetDB

```
  public void putAll(UserGroupDB newUserGroupMap) throws Exception {
    try {
      // loop through the cache of items to add, adding to this cache
      for (Fun.Tuple2<String, UserOrGroup> entry:newUserGroupMap.getCache()) {
        cache.add(entry);
        keys.add(keyToUog(entry.a));
      }
    }
    catch (Throwable t) {
      throw new Exception("UserGroupMapDB.mapDB-error", t, "MapDB error: putAll(UserGroupMapDB)");
    }
  }
```
 and at the end we perform the commit of the mapDB to disk

```
  public void commit() throws Exception {
    try {
      if (db != null)
        db.commit();
    }
    catch (Throwable t) {
      throw new Exception("UserGroupMapDB.mapDB-error", t, "MapDB error: commit()");
    }
  }
```

The first time the process works, but the second time fails in the commit method.

**Comment by amit-mhetre** [comment]
Hi,
I am also facing same issue.
I am using 1.0.6 mapDb version and java version "1.8.0_31".
I got this error while I try to update key-value pair.
Here is my stacktrace,

java.lang.AssertionError: unknown trans log instruction '0' at log offset: 16
	at org.mapdb.StoreWAL.replayLogFile(StoreWAL.java:858)
	at org.mapdb.StoreWAL.commit(StoreWAL.java:637)
	at org.mapdb.EngineWrapper.commit(EngineWrapper.java:94)
	at org.mapdb.DB.commit(DB.java:1595)

**Comment by jankotek** [comment]
@amit-mhetre There is number if bugs like this fixed in 1.0.7 release. Please upgrade
http://www.mapdb.org/changelog.html
